src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,462 @@
+    function move(address, address, uint256) external;
+    function frob(bytes32, address, address, address, int256, int256) external;
+    function grab(bytes32, address, address, address, int256, int256) external;
+    function fork(bytes32, address, address, int256, int256) external;
+    function suck(address, address, uint256) external;
+}
+
+interface EndLike {
+    function debt() external view returns (uint256);
+    function skim(bytes32, address) external;
+}
+
+interface DssDirectDepositPoolLike {
+    function king() external view returns (address);
+    function getMaxBar() external view returns (uint256);
+    function validTarget() external view returns (bool);
+    function calcSupplies(uint256) external view returns (uint256, uint256);
+    function deposit(uint256) external;
+    function withdraw(uint256) external;
+    function collect(address[] memory, uint256) external returns (uint256);
+    function transferShares(address, uint256) external returns (bool);
+    function assetBalance() external returns (uint256);
+    function shareBalance() external returns (uint256);
+    function convertToShares(uint256) external view returns(uint256);
+    function maxWithdraw() external view returns(uint256);
+    function cage() external;
+}
+
+contract DssDirectDepositHub {
+
+    // --- Auth ---
+    mapping (address => uint) public wards;
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+
+        emit Deny(usr);
+    }
+    modifier auth {
+        require(wards[msg.sender] == 1, "DssDirectDepositHub/not-authorized");
+        _;
+    }
+
+    struct Ilk {
+        DssDirectDepositPoolLike pool;
+        uint256                  tau; // Time until you can write off the debt [sec]
+        uint256                  culled;
+        uint256                  tic; // Timestamp when the pool is caged
+    }
+
+    ChainlogLike public immutable chainlog;
+    VatLike public immutable vat;
+    mapping (bytes32 => Ilk) public ilks;
+    TokenLike public immutable dai;
+    DaiJoinLike public immutable daiJoin;
+    uint256 public live = 1;
+
+    enum Mode{ NORMAL, MODULE_CULLED, MCD_CAGED }
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed ilk, bytes32 indexed what, address data);
+    event File(bytes32 indexed ilk, bytes32 indexed what, uint256 data);
+    event Wind(bytes32 indexed ilk, uint256 amount);
+    event Unwind(bytes32 indexed ilk, uint256 amount);
+    event Reap(bytes32 indexed ilk, uint256 amt);
+    event Collect(bytes32 indexed ilk, address indexed king, address[] assets, uint256 amt);
+    event Cage();
+    event Cage(bytes32 indexed ilk);
+    event Cull(bytes32 indexed ilk);
+    event Uncull(bytes32 indexed ilk);
+
+    constructor(address chainlog_) public {
```

Comment:
I think it ended up going into the pre-existing D3M but I'd really like to discuss removing the chainlog dependency from this in favor of just filing the `vow`, esp. now that this will be the only contract we're using for managing it.

By getting rid of the chainlog here we: 

1) remove a precondition for testing and secondary deployments
2) reduce the number of dynamic moving parts, which itself might itself have a critical issue
3) reduce the attack surface of the contract
4) provide clarity and intentionality in deployment and updates

The downside generally is that if we need to update the vow we have to make another `file` call, but I'm not sure the modest inconvenience of not having to make that call is worth the security tradeoffs.

Reply:
done https://github.com/makerdao/dss-direct-deposit/pull/31/commits/53152b7ee926cc12c00b754cf813562c67a93d6f